src/main/scala/scalafix/sbt/ScalafixPlugin.scala

```diff
@@ -96,12 +96,12 @@ object ScalafixPlugin extends AutoPlugin {
 
     def scalafixConfigSettings(config: Configuration): Seq[Def.Setting[_]] =
       Seq(
-        scalafix := scalafixInputTask(config).evaluated,
-        compile := Def.taskDyn {
+        (scalafix in config) := scalafixInputTask(config).evaluated,
+        (compile in config) := Def.taskDyn {
           val oldCompile =
-            compile.value // evaluated first, before the potential scalafix evaluation
+            (compile in config).value // evaluated first, before the potential scalafix evaluation
           if (scalafixOnCompile.value)
-            scalafix
+            (scalafix in config)
```

Comment:
we could wrap the `Seq` inside `inConfig(config)` instead of repeating the scope for each key

Reply:
addressed